### DISCUSSION TITLE
I want my trades to be 1% of the initial balance of the account.

### DISCUSSION THREAD

**Discussion by anderMartin92** [kernc/backtesting.py#737] [Q&A]
Hello, I want my trades to be 1% of the initial balance of the account. Even with 0 commissions the results vary a lot. Here I leave the formula with which I calculate 1%.

`money = 1000000

def my_size(sl, signal):
            ep = self.data.Close[-1]
            stop_percenti = trendPorcent(ep, (ep-sl), signal)
            position_size = (ep * (0.01 * money) / sl) / self.leverage 
            mysize = (position_size * 100 / money) / 100
            return float(mysize)

bt = Backtest(df, MyStrat, cash = money, margin=1/50, commission=.000)`

The results of PnL that it returns are these:
<img width="740" alt="Captura de Pantalla 2022-09-08 a las 1 34 43" src="https://user-images.githubusercontent.com/62348899/189001318-95fc4add-9664-4faa-baa6-0225958ce2e9.png">

If the initial balance is one million, they should generate a profit/loss of 10k (1%).

If I modify the commission to 0.002 it goes crazy and the result is this:

<img width="708" alt="Captura de Pantalla 2022-09-08 a las 1 41 52" src="https://user-images.githubusercontent.com/62348899/189001777-66c73be4-8bb5-4552-a58f-58cc95113780.png">

thank you for all!!

**Comment by kernc** [comment]
Try:
```py
class MyStrat(Strategy):
    def init(self):
        self._trade_max_cash = self.equity * .01  # 1% of initial cash

    def next(self):
        ...
            self.buy(..., size=int(self._trade_max_cash // self.data.Close[-1]))

**Reply by anderMartin92** [reply]
Hello, I have tried and having a million as an initial balance, it performs these small and unbalanced operations

<img width="768" alt="Captura de Pantalla 2022-09-09 a las 3 03 10" src="https://user-images.githubusercontent.com/62348899/189250978-b9044040-6306-49e0-a5ee-86e6c06977a0.png">

I have also tried removing the margin when launching the backtest. It has not worked.
bt = Backtest(df, MyStrat, cash = money, commission=.00)

Any other idea? I need to risk the same on each trade (1%). Greetings, thanks for everything.

**Reply by kernc** [reply]
8278 * 1.20767 ~= 10000, which _is_ 1% of 1e6.

The rest is on your strategy, the code for which you haven't posted.

**Reply by anderMartin92** [reply]
`if self.rsi[-1] < self.lower_bound  and len(self.trades)==0 and self.data.Close[-1] > self.data.Close[-2]:   
            sl1 = self.data.Low[-1]
            stoploss = self.data.Close[-1] - sl1
            tp1 = self.data.Close[-1] + stoploss * TPSLRatio
           
           self.buy(sl = sl1, tp = tp1, size=int(self._trade_max_cash / self.data.Close[-1]))`

If there are no open trades, the rsi is less than 30 and the last close is higher than the previous buy.

The SL that I really want 1% of my balance to be is the Low of the last candle.

If I don't specify the leverage, is it considered x1? Could it be related to our problem?
Thank you very much for your answer

**Reply by anderMartin92** [reply]
@kernc

**Reply by kernc** [reply]
> The SL that I really want 1% of my balance to be is the Low of the last candle.

I guess what you want then is something like:
```py
self.buy(..., size=int(self._trade_max_cash / stoploss))
```

> If I don't specify the leverage, is it considered x1? Could it be related to our problem? Thank you very much for your answer

Yes, [by default, `margin=1`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest), so 1/1 makes leverage 1x (i.e. 0).
